salt/utils/cloud.py

```diff
@@ -618,6 +618,14 @@ def bootstrap(vm_, opts=None):
     del event_kwargs['sudo_password']
     if 'password' in event_kwargs:
         del event_kwargs['password']
+    external_ip_dict = getattr(event_kwargs['vm_']['external_ip'], '__dict__', None)
```

Comment:
IMHO, this should be done in the gce driver.

Then we should add an call to pass event_kwargs through salt.utils.data.simple_types_filter, so that it removes any none serializable objects from the dictionary.